larpandoracontent/LArMonitoring/EventClusterValidationAlgorithm.cc

```diff
@@ -151,181 +249,681 @@ void EventClusterValidationAlgorithm::GetHitParents(
         }
     }
 
+    if (m_handleDeltaRays)
+    {
+        for (auto &[pCaloHit, parents] : hitParents)
+        {
+            parents.m_pMainMC = this->FoldPotentialDeltaRayTo(pCaloHit, parents.m_pMainMC);
```

Comment:
This might be more complex given the hit level decision making that I think this involves, rather than a pure MC consideration, but might this also fit into a precomputed folding?